test/eventHandlers/msg.ts

```diff
@@ -208,23 +211,54 @@ export namespace msg {
     }
 
     export namespace indexing {
+        export const request = {
+            requestId: 'testReqId',
+            functionsMetadataRequest: {
+                functionAppDirectory: testAppPath,
+            },
+        };
+
         export const receivedRequestLog = msg.receivedRequestLog('FunctionsMetadataRequest');
 
-        export function response(functions: rpc.IRpcFunctionMetadata[] = []): TestMessage {
+        export function response(
+            functions: rpc.IRpcFunctionMetadata[],
+            useDefaultMetadataIndexing: boolean
+        ): TestMessage {
             const response: rpc.IStreamingMessage = {
                 requestId: 'testReqId',
                 functionMetadataResponse: {
-                    useDefaultMetadataIndexing: functions.length === 0,
+                    useDefaultMetadataIndexing: useDefaultMetadataIndexing,
```

Comment:
Is `functions.length` no longer a valid proxy for `useDefaultMetadataIndexing`?

Reply:
No it is not. You can imagine a scenario where someone tries to register a function using model v4, but an error happens before the register finishes. In that case, we still want to treat that app as a v4 app for error handling purposes even though functions.length is still zero